### PR TITLE
Feature/new endpoint definition for server options

### PR DESCRIPTION
**Pull request by josemanuel167**
Add new endpoint definition and url prototype for remote options to questions

### CODE REVIEW

File: pepper-apis/docs/specification/src/endpoints/user.studies.activity.instance.question-options.yml
Comment (bskinner) [inline pepper-apis/docs/specification/src/endpoints/user.studies.activity.instance.question-options.yml]: Is this the correct response type for the 200 code?

**Comment by bskinner** [comment]
This looks like a good start, but the response schema for a 200 status code does not seem appropriate for this endpoint. That will need to be fleshed out before approval.

**Review by bskinner — CHANGES_REQUESTED** [review]

File: pepper-apis/docs/specification/src/endpoints/user.studies.activity.instance.question-options.yml
Comment (josemanuel167) [reply pepper-apis/docs/specification/src/endpoints/user.studies.activity.instance.question-options.yml]: Not sure. I think we could re use this class `ActivityInstanceSelectAnswerSubmission` and just change the file's name for a more appropiate one:

image

What do you think?

File: pepper-apis/docs/specification/src/endpoints/user.studies.activity.instance.question-options.yml
Comment (bskinner) [inline pepper-apis/docs/specification/src/endpoints/user.studies.activity.instance.question-options.yml]: There should also be a `q` query string parameter for filtering down the results- @MocanaAtBroad mentioned it in the most recent conversation between the three of us.

File: pepper-apis/docs/specification/src/endpoints/user.studies.activity.instance.question-options.yml
Comment (bskinner) [reply pepper-apis/docs/specification/src/endpoints/user.studies.activity.instance.question-options.yml]: `ActivityInstanceSelectAnswerSubmission` is not an appropriate template for this situation. First off, the `ActivityInstanceSelectAnswerSubmission` response is used when responding to the requests PATCHed to the `/answers` endpoint- we aren't doing that here. Further, at a minimum, the response from the `/options` endpoint should be returning a list of `PicklistOption` objects which match the request's `q` query parameter. Given that the `PicklistOption`s may also be part of a group, those groups should be included in the response as well.

Comment (bskinner) [reply pepper-apis/docs/specification/src/endpoints/user.studies.activity.instance.question-options.yml]: I'd suggest checking out the `getDrugSuggestions` and `getCancerSuggestions` endpoints for the general pattern we've used in the past.

Comment (josemanuel167) [reply pepper-apis/docs/specification/src/endpoints/user.studies.activity.instance.question-options.yml]: @bskinner Is it possible to do this :  `List<PicklistOption, PicklistGroup> results` in java?

image

Comment (bskinner) [reply pepper-apis/docs/specification/src/endpoints/user.studies.activity.instance.question-options.yml]: No- the [java.util.List](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html) class only supports a single type as a generic parameter. Structures like tuples are not supported in Java.

**Review by bskinner — DISMISSED** [review]

**Review by MocanaAtBroad — DISMISSED** [review]
Looks good!

**Review by MocanaAtBroad — DISMISSED** [review]

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**
